index.js

```diff
@@ -78,67 +81,34 @@ function lintFiles (files, opts, cb) {
   if (typeof files === 'string') files = [ files ]
   if (files.length === 0) files = DEFAULT_PATTERNS
 
-  // traverse filesystem
-  parallel(files.map(function (pattern) {
-    return function (cb) {
-      glob(pattern, {
-        cwd: opts.cwd,
-        ignore: opts._ignore,
-        nodir: true
-      }, cb)
-    }
-  }), function (err, results) {
+  deglob(files, opts, function (err, allFiles) {
     if (err) return cb(err)
-
-    // flatten nested arrays
-    var files = results.reduce(function (files, result) {
-      result.forEach(function (file) {
-        files.push(path.resolve(opts.cwd, file))
-      })
-      return files
-    }, [])
-
-    // de-dupe
-    files = uniq(files)
-
-    if (opts._gitignore) {
-      files = toRelative(opts.cwd, files)
-      if (os.platform() === 'win32') files = toUnix(files)
-      files = opts._gitignore.filter(files)
-      files = toAbsolute(opts.cwd, files)
-      if (os.platform() === 'win32') files = toWin32(files)
-    }
-
-    // undocumented – do not use (used by bin/cmd.js)
-    if (opts._onFiles) opts._onFiles(files)
+     // undocumented – do not use (used by bin/cmd.js)
+    if (opts._onFiles) opts._onFiles(allFiles)
 
     var result
     try {
-      result = new eslint.CLIEngine(opts._config).executeOnFiles(files)
+      result = new eslint.CLIEngine(opts._config).executeOnFiles(allFiles)
     } catch (err) {
       return cb(err)
     }
     return cb(null, result)
   })
+
 }
 
 function parseOpts (opts) {
   if (!opts) opts = {}
-  opts = extend(opts)
+  opts = extend(opts, DEGLOB_OPTIONS)
```

Comment:
Can you pass the options for `deglob` directly into that function rather than merging them in with the ones that standard requires? I think that's clearer. Remember to pass through options that are used by both, like `cwd` and `ignore`.